x-pack/plugins/enterprise_search/server/lib/indices/fetch_index_counts.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from '@kbn/core/server';
+
+export const fetchIndexCounts = async (client: IScopedClusterClient, indicesNames: string[]) => {
+  // TODO: is there way to batch this? Passing multiple index names or a pattern still returns a singular count
+  const countPromises = indicesNames.map(async (indexName) => {
```

Comment:
the count API does mention it supports multiple index names. I suspect if you pass in an array of index names into index, it should work in one call. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-count.html

Reply:
Sadly no, this just gives you one count across all indices. Not one count per index.